scripts/generate_docs.sh

```diff
@@ -0,0 +1,56 @@
+ #!/usr/bin/env bash
+
+set -eo pipefail
+
+mkdir -p ./tmp-swagger-gen
+
+# move the vendor folder to a temp dir so that go list works properly
+temp_dir="f29ea6aa861dc4b083e8e48f67cce"
+if [ -d vendor ]; then
+  mv ./vendor ./$temp_dir
+fi
+
+# Get the path of the cosmos-sdk repo from go/pkg/mod
+cosmos_sdk_dir=$(go list -f '{{ .Dir }}' -m github.com/cosmos/cosmos-sdk)
+
+# Get the path of the tokenfactory repo from go/pkg/mod
+tokenfactory_dir=$(go list -f '{{ .Dir }}' -m github.com/CosmWasm/token-factory)
+
+# move the vendor folder back to ./vendor
+if [ -d $temp_dir ]; then
+  mv ./$temp_dir ./vendor
+fi
+
+proto_dirs=$(find ./proto "$cosmos_sdk_dir"/proto "$tokenfactory_dir"/proto -path -prune -o -name '*.proto' -print0 | xargs -0 -n1 dirname | sort | uniq)
+for dir in $proto_dirs; do
+  # generate swagger files (filter query files)
+  query_file=$(find "${dir}" -maxdepth 1 \( -name 'query.proto' -o -name 'service.proto' \))
+  if [[ ! -z "$query_file" ]]; then
+    protoc \
```

Comment:
should we do a prerequisite check for `protoc` at the top of the script? Might be nice to tell users to install this before the script runs.

Reply:
Good call, fixed in baf772a4cbcaf8a153577a3b05852a3ea8935a12